linera-core/src/client/mod.rs

```diff
@@ -1941,11 +1946,11 @@ where
 
             let maybe_blob = {
                 let chain_state_view = self.chain_state_view().await?;
-                chain_state_view
-                    .manager
-                    .get()
-                    .pending_blobs
+                let manager = chain_state_view.manager.get();
+                manager
+                    .locked_published_blobs
```

Comment:
Actually, this is unnecessary: At the (only) call site, in `ChainClient::propose_block`, we only get into this situation when we already just looked up in the chain manager that we are re-proposing a locked block. In that case, just use the corresponding blobs that are already in the chain manager. Only call `read_local_blobs` in the `else` case, when we do `BlockProposal::new_initial`.

Reply:
Good catch!